### ISSUE TITLE
[Bug-gmx_MMPBSA]:  gmx make_ndx failed when querying index.ndx when using GMX2023

### ISSUE THREAD

**Issue by WoutVanEynde** [Valdes-Tresanco-MS/gmx_MMPBSA#335]
### Bug summary

When using GROMACS 2023 to calculate the PBSA I get an error regarding making the ndx file. I installed the software using the conda environment and the env.yml file.

### Terminal output

```bash
[INFO   ] Starting gmx_MMPBSA v1.5.7
[INFO   ] Command-line
  mpirun -np 12 gmx_MMPBSA MPI -i mmpbsa.in -cs md.tpr -ci index2.ndx -cg 1 12 -ct md_noPBC_whole_nojump_center_mol_com_fit.xtc -cp topol_complex.top -o FINAL_RESULTS_MMPBSA.dat -eo FINAL_RESULTS_MMPBSA.csv -do FINAL_DECOMP_MMPBSA.dat -deo FINAL_DECOMP_MMPBSA.csv

[INFO   ] Checking mmpbsa.in input file...
[INFO   ] Checking mmpbsa.in input file...Done.

[INFO   ] Checking external programs...
[INFO   ] cpptraj found! Using /home/wout/anaconda3/envs/gmxMMPBSA/bin/cpptraj
[INFO   ] tleap found! Using /home/wout/anaconda3/envs/gmxMMPBSA/bin/tleap
[INFO   ] parmchk2 found! Using /home/wout/anaconda3/envs/gmxMMPBSA/bin/parmchk2
[INFO   ] sander found! Using /home/wout/anaconda3/envs/gmxMMPBSA/bin/sander
[INFO   ] Using GROMACS version > 5.x.x!
[INFO   ] gmx found! Using /usr/local/gromacs/bin/gmx
[INFO   ] Checking external programs...Done.

[INFO   ] Building AMBER topologies from GROMACS files...
[INFO   ] Get PDB files from GROMACS structures files...
[INFO   ] Making gmx_MMPBSA index for complex...
[ERROR  ] MMPBSA_Error /usr/local/gromacs/bin/gmx make_ndx failed when querying index2.ndx.
           Check the gmx_MMPBSA.log file to report the problem.
  File "/home/wout/anaconda3/envs/gmxMMPBSA/bin/gmx_MMPBSA", line 8, in <module>
    sys.exit(gmxmmpbsa())
  File "/home/wout/anaconda3/envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/app.py", line 98, in gmxmmpbsa
    app.make_prmtops()
  File "/home/wout/anaconda3/envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/main.py", line 548, in make_prmtops
    self.FILES.mutant_receptor_prmtop, self.FILES.mutant_ligand_prmtop) = maketop.buildTopology()
  File "/home/wout/anaconda3/envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/make_top.py", line 116, in buildTopology
    self.gmx2pdb()
  File "/home/wout/anaconda3/envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/make_top.py", line 254, in gmx2pdb
    GMXMMPBSA_ERROR('%s failed when querying %s' % (' '.join(self.make_ndx), self.FILES.complex_index))
  File "/home/wout/anaconda3/envs/gmxMMPBSA/lib/python3.9/site-packages/GMXMMPBSA/exceptions.py", line 169, in __init__
    raise exc(msg + '\nCheck the gmx_MMPBSA.log file to report the problem.')
MMPBSA_Error: /usr/local/gromacs/bin/gmx make_ndx failed when querying index2.ndx
Check the gmx_MMPBSA.log file to report the problem.
Error occurred on rank 0.
Exiting. All files have been retained.
application called MPI_Abort(MPI_COMM_WORLD, 1) - process 0
```

### gmx_MMPBSA.log

[gmx_MMPBSA.log](https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA/files/10765129/gmx_MMPBSA.log)

### Operating system

Linux

### gmx_MMPBSA Version

v1.5.7

### Python version

3.9.15

### Installation

None

**Comment by marioernestovaldes** [comment]
Check this thread... it is a bug that has been fixed in the development version...

https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA/issues/327#issuecomment-1426622722

**Comment by Allzzw** [comment]
The above problems (gmx make_ndx failed when querying index.ndx when using GMX2023) have been solved after the code (amber.python -m pip install git+https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA) is updated to the development version.
Then, the new problem:

pro@Thermal-Throttling-Mac test % gmx_MMPBSA -O -i mmpbsa.in -cs md_0_1.tpr -ci index.ndx -cg 18 19 -ct md_0_1_noPBC.xtc

[INFO   ] Starting gmx_MMPBSA v1.5.7+19.gc3ce167
[INFO   ] Command-line
  gmx_MMPBSA -O -i mmpbsa.in -cs md_0_1.tpr -ci index.ndx -cg 18 19 -ct md_0_1_noPBC.xtc

[INFO   ] Checking mmpbsa.in input file...
[INFO   ] Checking mmpbsa.in input file...Done.

[INFO   ] Checking external programs...
[INFO   ] cpptraj found! Using /usr/local/Cellar/dir/amber22/bin/cpptraj
[INFO   ] tleap found! Using /usr/local/Cellar/dir/amber22/bin/tleap
[INFO   ] parmchk2 found! Using /usr/local/Cellar/dir/amber22/bin/parmchk2
[INFO   ] sander found! Using /usr/local/Cellar/dir/amber22/bin/sander
[INFO   ] Using GROMACS version > 5.x.x!
[INFO   ] gmx found! Using /usr/local/bin/gmx
[INFO   ] Checking external programs...Done.

[INFO   ] Building AMBER topologies from GROMACS files...
[INFO   ] Get PDB files from GROMACS structures files...
[INFO   ] Making gmx_MMPBSA index for complex...
[INFO   ] Normal Complex: Saving group a_1-4291_a_4292-6221 (18_19) in _GMXMMPBSA_COM_index.ndx file as _GMXMMPBSA_COM.pdb
[ERROR  ] MMPBSA_Error /usr/local/bin/gmx trjconv failed when querying md_0_1_noPBC.xtc.
           Check the gmx_MMPBSA.log file to report the problem.
  File "/usr/local/Cellar/dir/amber22/miniconda/bin/gmx_MMPBSA", line 8, in <module>
    sys.exit(gmxmmpbsa())
  File "/usr/local/Cellar/dir/amber22/miniconda/lib/python3.10/site-packages/GMXMMPBSA/app.py", line 98, in gmxmmpbsa
    app.make_prmtops()
  File "/usr/local/Cellar/dir/amber22/miniconda/lib/python3.10/site-packages/GMXMMPBSA/main.py", line 548, in make_prmtops
    self.FILES.mutant_receptor_prmtop, self.FILES.mutant_ligand_prmtop) = maketop.buildTopology()
  File "/usr/local/Cellar/dir/amber22/miniconda/lib/python3.10/site-packages/GMXMMPBSA/make_top.py", line 116, in buildTopology
    self.gmx2pdb()
  File "/usr/local/Cellar/dir/amber22/miniconda/lib/python3.10/site-packages/GMXMMPBSA/make_top.py", line 283, in gmx2pdb
    GMXMMPBSA_ERROR('%s failed when querying %s' % (' '.join(comprog), self.FILES.complex_trajs[0]))
  File "/usr/local/Cellar/dir/amber22/miniconda/lib/python3.10/site-packages/GMXMMPBSA/exceptions.py", line 169, in __init__
    raise exc(msg + '\nCheck the gmx_MMPBSA.log file to report the problem.')
MMPBSA_Error: /usr/local/bin/gmx trjconv failed when querying md_0_1_noPBC.xtc
Check the gmx_MMPBSA.log file to report the problem.
Exiting. All files have been retained.


**gmx_MMPBSA.log**
[gmx_MMPBSA.log](https://github.com/Valdes-Tresanco-MS/gmx_MMPBSA/files/10767897/gmx_MMPBSA.log)

Operating system
MacOS

gmx_MMPBSA Version
v1.5.7

Python version
Python 3.10.9

Installation
Gromacs2023 from Homebrew
Amber22 from http://ambermd.org/InstMacOS.php

**Comment by Valdes-Tresanco-MS** [comment]
Oh, the problem is that MacOS doesn't recognize the `-e` flag of the echo command. I will fix it as soon as possible.

**Comment by Valdes-Tresanco-MS** [comment]
@Allzzw I have fixed the bug (ba0a71ddb2e792f7d585c694b13fa926d07eada9), but I have no way to test it on macOS. Can you test if it works for you?

**Comment by Allzzw** [comment]
Thank you for your help. I have successfully run the binding energy calculation of gmx_MMPBSA on my Mac.
